src/imitation/data/wrappers.py

```diff
@@ -1,15 +1,86 @@
 """Environment wrappers for collecting rollouts."""
 
+import os
+import shutil
+import tempfile
+import uuid
 from typing import List, Optional, Sequence, Tuple
 
 import gymnasium as gym
+import cv2
 import numpy as np
 import numpy.typing as npt
 from stable_baselines3.common.vec_env import VecEnv, VecEnvWrapper
 
 from imitation.data import rollout, types
 
 
+class RenderImageInfoWrapper(gym.Wrapper):
+    """Saves render images to `info`.
+
+    Can be very memory intensive for large render images.
+    Use `scale_factor` to reduce render image size.
+    If you need to preserve the resolution and memory
+    runs out, you can activate `use_file_cache` to save
+    rendered images and instead put their path into `info`.
+    """
+
+    def __init__(
+        self,
+        env: gym.Env,
+        scale_factor: float = 1.0,
+        use_file_cache: bool = False,
+    ):
+        """Builds RenderImageInfoWrapper.
+
+        Args:
+            env: Environment to wrap.
+            scale_factor: scales rendered images to be stored.
+            use_file_cache: whether to save rendered images to disk.
+        """
+        super().__init__(env)
+        self.scale_factor = scale_factor
+        self.use_file_cache = use_file_cache
+        if self.use_file_cache:
+            self.file_cache = tempfile.mkdtemp("imitation_RenderImageInfoWrapper")
+
+    def step(self, action):
+        obs, rew, done, info = self.env.step(action)
```

Comment:
This wrapper hasn't been adjusted for the new gymnasium api yet right?

Reply:
that's right, thanks for the heads up! We also still need to add tests for the wrapper.

Reply:
Hey @timokau, the wrapper is adjusted to the gymnasium api and we tested it briefly. It should work. hope this helps

Reply:
Thanks Marvin!